lazyscribe/experiment.py

```diff
@@ -365,13 +363,14 @@ def log_test(self, name: str, description: Optional[str] = None) -> Iterator[Tes
             The :py:class:`lazyscribe.test.Test` dataclass.
         """
         test = Test(name=name, description=description)
+
+        self.last_updated = utcnow()
+        self.dirty = True

```

Comment:
What's the rational behind moving this call? I put it after the `yield` so we don't label the `Experiment` object as `dirty` if there's an exception raised within the context handler.

Reply:
> if there's an exception raised within the context handler.

But handler can mutate `test` _before_ raising an exception, can't it?

Reply:
Yes but the test is not associated with the experiment until after the `yield`. If an exception is thrown while the context handler is open, the experiment will remain unchanged (and it's status as `dirty` shouldn't change)

Reply:
Oh, right. I'll repush.